Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift

```diff
@@ -0,0 +1,119 @@
+        self.style = style
+        self.initialCountry = initialCountry
+        self.initialRegionCode = initialRegionCode
+        self.delegate = delegate
+    }
+    
+    func getCell(for row: Int, delegate: UIViewController? = nil) -> UITableViewCell {
+        if row == 0 {
+            return getHeaderCell(delegate: delegate as? BillingFormHeaderCellDelegate)
+        }
+        return getFieldCell(for: row, delegate: delegate as? BillingFormTextFieldCellDelegate)
+    }
+     
+    private func getHeaderCell(delegate: BillingFormHeaderCellDelegate?) -> UITableViewCell {
+        var style = style.header
+        updateHeaderView(with: &style)
+        let cell = BillingFormHeaderCell(style: style, delegate: delegate)
+        self.editDelegate = cell
+        return cell
+    }
+    
+    private func getFieldCell(for row: Int, delegate: BillingFormTextFieldCellDelegate?) -> UITableViewCell {
+        guard let type = BillingFormCellType(rawValue: row) else { return UITableViewCell()}
+        var style = style.fields[row - 1]
+        updateErrorView(with: &style, type: type)
+        updateTextField(with: &style, type: type)
+        return BillingFormTextFieldCell(type: type, style: style, delegate: delegate)
+    }
+    
+    private func updateHeaderView(with style: inout BillingFormHeaderCellStyle) {
+        style.doneButton.isEnabled = textValueOfCellType.values.count == self.style.fields.count
+    }
+    
+    private func updateErrorView(with style: inout BillingFormTextFieldCellStyle, type: BillingFormCellType) {
+        guard let hasError = errorFlagOfCellType[type] else { return }
+        style.error.isHidden = !hasError
+    }
+    
+    private func updateTextField(with style: inout BillingFormTextFieldCellStyle, type: BillingFormCellType) {
+        guard let text = textValueOfCellType[type] else { return }
+        style.textfield.text = text
+    }
+}
+
+extension DefaultBillingFormViewModel: BillingFormViewControllerDeleagte {
+    func textFieldIsChanged(textField: BillFormTextField, replacementString: String) {
+        if !replacementString.isEmpty {
+            textValueOfCellType[textField.type] = replacementString
+        }
+
+        if textValueOfCellType.values.count == style.fields.count {
+            editDelegate?.didFinishEditingBillingForm(successfully: !errorFlagOfCellType.values.allSatisfy({$0}))
+        }
+    }
+    
+    func tableView(estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        UITableView.automaticDimension
+    }
+    
+    func tableView(numberOfRowsInSection section: Int) -> Int {
+        style.fields.count + 1
+    }
+    
+    func tableView(cellForRowAt indexPath: IndexPath, sender: UIViewController) -> UITableViewCell {
+        getCell(for: indexPath.row, delegate: sender)
+    }
+    
+    func doneButtonIsPressed(sender: UIViewController) {
+        let countryCode = "44" //"\(addressView.phoneInputView.phoneNumber?.countryCode ?? 44)"
```

Comment:
I think there is something wrong on this line but can't tell for certain how it should be